src/cascadia/TerminalSettingsEditor/Extensions.cpp

```diff
@@ -0,0 +1,292 @@
+        fragmentExtensions.reserve(settings.FragmentExtensions().Size());
+
+        // these vectors track components all extensions successfully added
+        std::vector<Editor::FragmentProfileViewModel> profilesModifiedTotal;
+        std::vector<Editor::FragmentProfileViewModel> profilesAddedTotal;
+        std::vector<Editor::FragmentColorSchemeViewModel> colorSchemesAddedTotal;
+        for (const auto&& fragExt : settings.FragmentExtensions())
+        {
+            const auto extensionEnabled = GetExtensionState(fragExt.Source());
+
+            // these vectors track everything the current extension attempted to bring in
+            std::vector<Editor::FragmentProfileViewModel> currentProfilesModified;
+            std::vector<Editor::FragmentProfileViewModel> currentProfilesAdded;
+            std::vector<Editor::FragmentColorSchemeViewModel> currentColorSchemesAdded;
+
+            for (const auto&& entry : fragExt.ModifiedProfilesView())
+            {
+                // Ensure entry successfully modifies a profile before creating and registering the object
+                if (const auto& deducedProfile = _settings.FindProfile(entry.ProfileGuid()))
+                {
+                    auto vm = winrt::make<FragmentProfileViewModel>(entry, fragExt, deducedProfile);
+                    currentProfilesModified.push_back(vm);
+                    if (extensionEnabled)
+                    {
+                        profilesModifiedTotal.push_back(vm);
+                    }
+                }
+            }
+
+            for (const auto&& entry : fragExt.NewProfilesView())
+            {
+                // Ensure entry successfully points to a profile before creating and registering the object.
+                // The profile may have been removed by the user.
+                if (const auto& deducedProfile = _settings.FindProfile(entry.ProfileGuid()))
+                {
+                    auto vm = winrt::make<FragmentProfileViewModel>(entry, fragExt, deducedProfile);
+                    currentProfilesAdded.push_back(vm);
+                    if (extensionEnabled)
+                    {
+                        profilesAddedTotal.push_back(vm);
+                    }
+                }
+            }
+
+            for (const auto&& entry : fragExt.ColorSchemesView())
+            {
+                for (const auto& schemeVM : _colorSchemesPageVM.AllColorSchemes())
+                {
+                    if (schemeVM.Name() == entry.ColorSchemeName())
+                    {
+                        auto vm = winrt::make<FragmentColorSchemeViewModel>(entry, fragExt, schemeVM);
+                        currentColorSchemesAdded.push_back(vm);
+                        if (extensionEnabled)
+                        {
+                            colorSchemesAddedTotal.push_back(vm);
+                        }
+                    }
+                }
+            }
```

Comment:
I'm only displaying the ones that were added successfully and point to something that already exists.

Long-term, do we want to do more here? Perhaps show the ones that weren't added successfully? I'm just not sure what the next steps would be and _how_ we would want to approach this issue.